plugins/react.js

```diff
@@ -177,6 +177,8 @@ module.exports = {
     'react/no-will-update-set-state': 2,
     // enforce ES5 or ES6 class for React Components
     'react/prefer-es6-class': 2,
+    // ensure only exact prop definitions are used
+    'react/prefer-exact-props': 1,
```

Comment:
I added the rule as a warning, but open to changing it to an error if that is our team opinion.

Reply:
I'm trying to understand the impact of this check with respect to prop spreading. What happens if you `yarn link` this change into `react-components` and then run `yarn lint:js` over there?

Reply:
Thanks for that call out. I tried this and everything currently produces a warning because we don't use exact propTypes, which if we do, might very well conflict with prop spreading. However seeing as we don't forbid JSX prop spreading, I'm thinking this one should be turned off really instead of being a warning. I'll update it to be a `0` unless there are objections.

Reply:
I like the initial instinct for clean code. But I think `0` is the right call for Garden here.